sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -352,13 +352,16 @@ object IntervalUtils {
           val sign = finalSign(firstSign, secondSign)
           unit.toUpperCase(Locale.ROOT) match {
             case "DAY" if suffix == null && value.length <= 9 && checkTargetType(DT.DAY, DT.DAY) =>
-              sign * value.toLong * MICROS_PER_DAY
+              val days = toLongWithRange(dayStr, value, 0, MAX_DAY)
+              sign * days * MICROS_PER_DAY
```

Comment:
Could you re-use the existing function which checks arithmetic ops too:
```suggestion
              toDTInterval(value, "0", "0", "0", sign)
```

Reply:
Done. But for the hours entry can't use this, because of hours range is not [0, 23] but a large number.